### PR TITLE
Pods 8124 - Prevent User from Clicking the Declaration Button Multiple Times During Event Report Submission

### PR DESCRIPTION
**Pull request by jamdade-sharad**

### CODE REVIEW

File: app/connectors/EventReportingConnector.scala [hunk 1]
```diff
@@ -107,7 +109,7 @@ class EventReportingConnector @Inject()(
   }
 
   def submitReport(pstr: String, ua: UserAnswers, version: String)
-                  (implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+                  (implicit headerCarrier: HeaderCarrier): Future[Result] = {
```
Comment (PavelVjalicin) [inline app/connectors/EventReportingConnector.scala:113]: Change submitReport return type to Future[HttpResponse] and handle via controller.

File: app/connectors/EventReportingConnector.scala [hunk 2]
```diff
@@ -129,7 +132,7 @@ class EventReportingConnector @Inject()(
   }
 
   def submitReportEvent20A(pstr: String, ua: UserAnswers, version: String)
-                          (implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+                          (implicit headerCarrier: HeaderCarrier): Future[Result] = {
```
Comment (PavelVjalicin) [inline app/connectors/EventReportingConnector.scala:135]: Change submitReportEvent20A return type to Future[HttpResponse] and handle via controller.

File: app/services/SubmitService.scala [hunk 3]
```diff
@@ -36,10 +37,16 @@ class SubmitService @Inject()(
                   (implicit ec: ExecutionContext, headerCarrier: HeaderCarrier): Future[Result] = {
     ua.get(VersionInfoPage) match {
       case Some(VersionInfo(version, Compiled)) =>
-        eventReportingConnector.submitReport(pstr, ua, version.toString).flatMap { _ =>
-          val updatedUA = ua.setOrException(VersionInfoPage, VersionInfo(version, Submitted), nonEventTypeData = true)
-          userAnswersCacheConnector.save(pstr, updatedUA).map { _ => Ok }
+        eventReportingConnector.submitReport(pstr, ua, version.toString).flatMap { response =>
```
Comment (PavelVjalicin) [inline app/services/SubmitService.scala:40]: eventReportingConnector.submitReport should return Future[HttpResponse] and this function also should return Future[HttpResponse] and should be handled via controller.

**Review by PavelVjalicin — CHANGES_REQUESTED** [review]
Otherwise, looks good.

File: app/connectors/EventReportingConnector.scala (hunk 2 above)
Comment (jamdade-sharad) [reply app/connectors/EventReportingConnector.scala:135]: After discussion with Pavel, these comments will be addressed in [PODS-8697](https://jira.tools.tax.service.gov.uk/browse/PODS-8697) as this is a substantial refactor which will require QA testing again.

**Comment by PavelVjalicin** [comment]
Conflict needs resolving.

**Review by PavelVjalicin — APPROVED** [review]

**Pull request merged**